frontend/src/lib/components/PropertyFilters/PropertyFilter.js

```diff
@@ -2,99 +2,80 @@ import React, { Component } from 'react'
 import Select from 'react-select'
 import { CloseButton, selectStyle } from '../../utils'
 import { PropertyValue } from './PropertyValue'
-import PropTypes from 'prop-types'
+import { useValues, useActions } from 'kea'
+import { propertyFilterLogic } from './propertyFilterLogic'
 
-export class PropertyFilter extends Component {
-    constructor(props) {
-        super(props)
-
-        this.state = {}
+export function PropertyFilter({ index, endpoint, onChange }) {
+    const { properties, filters } = useValues(propertyFilterLogic({ onChange }))
+    const { set, remove } = useActions(propertyFilterLogic({ onChange }))
+    let item = filters[index]
+    let key = Object.keys(item)[0] ? Object.keys(item)[0].split('__') : []
+    let value = Object.values(item)[0]
+    let operatorMap = {
+        null: 'equals',
+        is_not: "doesn't equal",
+        icontains: 'contains',
+        not_icontains: "doesn't contain",
+        gt: 'greater than',
+        lt: 'lower than',
     }
-    render() {
-        let { properties, index, item, onSet, onRemove, endpoint } = this.props
-        let key = Object.keys(item)[0] ? Object.keys(item)[0].split('__') : []
-        let value = Object.values(item)[0]
-        let operatorMap = {
-            null: 'equals',
-            is_not: "doesn't equal",
-            icontains: 'contains',
-            not_icontains: "doesn't contain",
-            gt: 'greater than',
-            lt: 'lower than',
-        }
-        return (
-            <div className="row" style={{ margin: '0.5rem -15px' }}>
-                <div className="col-3" style={{ paddingRight: 0 }}>
+    return (
+        <div className="row" style={{ margin: '0.5rem -15px' }}>
+            <div className="col-3" style={{ paddingRight: 0 }}>
+                {properties && (
                     <Select
                         options={properties}
                         style={{ width: 200 }}
                         value={[{ label: key[0], value: key[0] }]}
                         isLoading={!properties}
                         placeholder="Property key"
                         onChange={item =>
-                            onSet(
-                                item.value + (key[1] ? '__' + key[1] : ''),
-                                item.value != key[0] ? '' : value
-                            )
+                            set(index, item.value + (key[1] ? '__' + key[1] : ''), item.value != key[0] ? '' : value)
```

Comment:
Also, I recommend getting into the habit of using `!==` instead of `!=` and `===` instead of `==`... unless this was intentional. But even then my linter would complain about it.